Client/Application/AppDelegate.swift

```diff
@@ -64,6 +64,8 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UIViewControllerRestorati
         
         HttpsEverywhereStats.shared.startLoading()
         
+        //Moves Coredata sqlite file from Documents dir to application support dir.
+        DataController.shared.migrateToNewPathIfNeeded()
```

Comment:
We should probably move this into the migration, and just let the migration call below handle this if possible. Is there a timing issue or something?

Best not to require different migrations to happen at different places.

Reply:
So I need the db migration to run before any db access. If I can enforce that no one will ever write anything DB before migration code then we can move it there. Any ideas we can do that?

Reply:
Unless there are issues right now, I think this is relatively safe to assume. Migrations should generally always happen before DB access, and hasn't been an issue this far (famous last words, right?). 

Also, it would only matter if we accessed the DB before someone had this migration take place, so after a couple months, this migration will likely not be very useful/important.

If we have genuine concerns about this, we should consider moving the entire migration process up.

Reply:
As discussed in DM, The placement of the db migration is curcial because of ` !DataController.shared.storeExists() ` check. Create a facade for the DB migration in Migration Code though.